DS13/code/modules/overmap/combat.dm

```diff
@@ -0,0 +1,507 @@
+	for(var/obj/structure/overmap_component/XX in powered_components)
+		if(istype(XX, /obj/structure/overmap_component/plasma_relay))
+			var/obj/structure/overmap_component/plasma_relay/PS = XX
+			if(PS.supplying == OM.damage_sector && PS.obj_integrity >= 40)
+				PS.take_damage(mod)
+
+/obj/structure/overmap/proc/explode()
+	qdel(shield_overlay)
+	qdel(shields)
+	SpinAnimation(1000,1000)
+	weapon_power = 0
+	shield_power = 0
+	engine_power = 0
+	power_slots = 0
+	movement_block = TRUE
+	remove_control()
+	send_sound_crew('DS13/sound/effects/damage/ship_explode.ogg')
+	addtimer(CALLBACK(src, .proc/core_breach_finish), 450)
+	for(var/mob/A in operators)
+		to_chat(A, "<span class='cult'><font size=3>Your ship has been destroyed!")
+		if(A.remote_control)
+			A.remote_control.forceMove(get_turf(A))
+	core_breach()
+
+/obj/structure/overmap/proc/core_breach()
+	if(!main_overmap)
+		if(!linked_area)
+			find_area()
+		for(var/mob/player in linked_area)
+			SEND_SOUND(player, 'DS13/sound/effects/damage/corebreach.ogg')
+		return
+	for(var/mob/player in GLOB.player_list)
+		if(is_station_level(player.z))
+			SEND_SOUND(player, 'DS13/sound/effects/damage/corebreach.ogg')
+
+/obj/structure/overmap/proc/core_breach_finish()
+	if(main_overmap)
+		Cinematic(CINEMATIC_NUKE_WIN,world)
+		SSticker.mode.check_finished(TRUE)
+		SSticker.force_ending = 1
+		qdel(src)
+		return
+	for(var/I = 0, I <= 11, I++) //If it's not a game-ender. Blow the shit out of the ship map
+		var/turf/T = pick(get_area_turfs(linked_area))
+		explosion(T,10,10,10)
+	qdel(src)
+
+/obj/structure/overmap/proc/special_fx_targeted(var/shields_absorbed) //This ship isn't the main overmap, so find the area we want and apply damage to it
+	if(!linked_area)
+		find_area()
+	for(var/mob/player in linked_area)
+		shake_camera(player, 1,3)
+		if(shields_absorbed)
+			if(prob(50))
+				var/sound/shieldhit = pick('DS13/sound/effects/damage/shield_hit.ogg','DS13/sound/effects/damage/shield_hit2.ogg')
+				SEND_SOUND(player, shieldhit)
+			continue
+		var/sound/S = pick('DS13/sound/effects/damage/shiphit.ogg','DS13/sound/effects/damage/shiphit2.ogg','DS13/sound/effects/damage/shiphit3.ogg','DS13/sound/effects/damage/creak1.ogg','DS13/sound/effects/damage/creak2.ogg')
+		SEND_SOUND(player, S)
+		if(prob(40))
+			var/turf/T = pick(get_area_turfs(target))
+			new /obj/effect/temp_visual/explosion_telegraph(T)
+		if(prob(10))
+			var/turf/T = pick(get_area_turfs(linked_area))
+			T.atmos_spawn_air("plasma=20;TEMP=1000")
+
+
+/obj/structure/overmap/proc/show_damage(var/amount, var/shields_absorbed) //Flash up numbers showing how much damage we just took
+	if(amount > 100)
+		amount = 100 //We can only show up to 100 currently
+	//amount = round(10) //Get it to the nearest 10
+	if(shields_absorbed)
+		switch(amount)
+			if(0 to 9) new /obj/effect/temp_visual/damage_indicator/shield/zero (src.loc) //Shields took the hit, show a blue number
```

Comment:
please don't put the effects on the same line as the if

Reply:
Correct it also put the comments on the line below its really too long